### PR TITLE
Update camera follow implementation

### PR DESCRIPTION
**Pull request by Hanson-Sun**

### CODE REVIEW

File: Assets/Scripts/SceneUtility/CameraFollowPlayer.cs [hunk 1]
```diff
@@ -5,12 +5,21 @@
 public class CameraFollowPlayer : MonoBehaviour
 {
     public GameObject player;
-    public Vector2 offset;
+    public Vector3 offset;
+    public float smoothSpeed = 3.5f;
```
Comment (leo-cf-tian) [inline Assets/Scripts/SceneUtility/CameraFollowPlayer.cs:9]: Please make variables const whenever possible. Note that consts should come before all other fields.

File: Assets/Scripts/SceneUtility/CameraFollowPlayer.cs [hunk 2]
```diff
@@ -5,12 +5,21 @@
...
+    public Vector3 difference;
+
+    void Start()
+    {
+        difference = transform.position - player.transform.position;
+        offset.y = -2f;
```
Comment (leo-cf-tian) [inline Assets/Scripts/SceneUtility/CameraFollowPlayer.cs:15]: Just curious what this is for.

**Review by leo-cf-tian — CHANGES_REQUESTED** [review]

File: Assets/Scripts/SceneUtility/CameraFollowPlayer.cs (hunk 2 above)
Comment (Hanson-Sun) [reply Assets/Scripts/SceneUtility/CameraFollowPlayer.cs:15]: if its not there it wont work 😰

File: Assets/Scripts/SceneUtility/CameraFollowPlayer.cs (hunk 1 above)
Comment (Hanson-Sun) [reply Assets/Scripts/SceneUtility/CameraFollowPlayer.cs:9]: only smoothspeed should be a const

Comment (anjx123) [reply Assets/Scripts/SceneUtility/CameraFollowPlayer.cs]: If you make the smooth speed public, you can set it in the editor and so you dont have to set it inside the script

Comment (leo-cf-tian) [reply Assets/Scripts/SceneUtility/CameraFollowPlayer.cs]: that also works, depends on you implementation.

**Review by leo-cf-tian — APPROVED** [review]

**Pull request merged**